artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQDestination.java

```diff
@@ -244,22 +262,32 @@ public static ActiveMQTemporaryTopic createTemporaryTopic(String address) {
    protected ActiveMQDestination(final String address,
                                  final TYPE type,
                                  final ActiveMQSession session) {
-      this.simpleAddress = SimpleString.toSimpleString(address);
-
-      this.thetype = type;
-
-      this.session = session;
+      this(SimpleString.toSimpleString(address), type, session);
+   }
 
-      this.temporary = TYPE.isTemporary(type);
+   protected ActiveMQDestination(final SimpleString address,
+                                 final TYPE type,
+                                 final ActiveMQSession session) {
+      this(address, address != null ? address.toString() : null, type, session);
+   }
 
-      this.queue = TYPE.isQueue(type);
+   protected ActiveMQDestination(final String address,
+                                 final String name,
+                                 final TYPE type,
+                                 final ActiveMQSession session) {
+      this(SimpleString.toSimpleString(address), name, type, session);
    }
 
    protected ActiveMQDestination(final SimpleString address,
```

Comment:
Can this please be marked as deprecated and the main or super constructor just care for simple address.

Reply:
e.g.
```
   protected ActiveMQDestination(final String address,
                                 final TYPE type,
                                 final ActiveMQSession session) {
      this(SimpleString.toSimpleString(address), type, session);
   }

   protected ActiveMQDestination(final SimpleString address,
                                 final TYPE type,
                                 final ActiveMQSession session) {
      setSimpleAddress(address);

      this.thetype = type;

      this.session = session;

      this.temporary = TYPE.isTemporary(type);

      this.queue = TYPE.isQueue(type);
   }

   @Deprecated
   protected ActiveMQDestination(final String address,
                                 final String name,
                                 final TYPE type,
                                 final ActiveMQSession session) {
      this(SimpleString.toSimpleString(address), name, type, session);
   }

   @Deprecated
   protected ActiveMQDestination(final SimpleString address,
                                 final String name,
                                 final TYPE type,
                                 final ActiveMQSession session) {
      this(address, type, session);
      this.name = name;
   }
```